### PR TITLE
Add codespell config and workflow to detect new typos, fix some already found typos

### PR DESCRIPTION
**Pull request by yarikoptic**
I love dependabot's service so decided to contribute while looking through its codebase.  I have contributed codespell integration to a good number of projects and generally it is quite a nice help.  Here in dependabot though it might be a bit trickier since I think some "typos" are there for a reason. E.g. may be `"holiday defintions"` is there to be matched somewhere etc. I would appreciate someone going through the diff and pointing to false positives which we should ignore

TODOs
- [x] identify and ignore false positives

### CODE REVIEW

**Comment by deivid-rodriguez** [comment]
Hello! I'm sorry nobody got back to you before, but I think this is cool, thanks so much! Are you planing to move on with this project?

**Comment by yarikoptic** [comment]
> Are you planing to move on with this project?

I would if there is a blessing and interest from developers.

**Comment by deivid-rodriguez** [comment]
I am interested, and will happily review it!

**Comment by yarikoptic** [comment]
> I am interested, and will happily review it!

ok, I have rebased, reviewed more, added more skips, and reran fixing. Looked clean upon my review but please check the diff now for false positives etc.

File: cargo/spec/dependabot/cargo/file_parser_spec.rb [hunk 1]
```diff
@@ -521,8 +521,8 @@
         end
       end
 
-      context "that is unparseable" do
-        let(:manifest_fixture_name) { "unparseable" }
+      context "that is unparsable" do
+        let(:manifest_fixture_name) { "unparsable" }
```
Comment (deivid-rodriguez) [inline cargo/spec/dependabot/cargo/file_parser_spec.rb:525]: This is an actual file, so if we make the switch we also need to rename the file. According to google, both seem correct and unparseable more commonly used, so maybe we can keep it as is and add an ignore for it.

Comment (yarikoptic) [reply cargo/spec/dependabot/cargo/file_parser_spec.rb]: ok, ignored, force-pushed rerun of codespell

**Comment by deivid-rodriguez** [comment]
Just a couple more failures, can you have a look? We're almost there!

**Comment by yarikoptic** [comment]
```Run set -o pipefail
    cli | 2023/12/20 16:00:17 pulling image: ghcr.io/github/dependabot-update-job-proxy/dependabot-update-job-proxy:latest
    cli | 2023/12/20 16:00:24 failed to pull ghcr.io/github/dependabot-update-job-proxy/dependabot-update-job-proxy:latest: Error response from daemon: received unexpected HTTP status: 502 Bad Gateway
16
Error: Process completed with exit code 1.
```

looks like possible intermittent failure... I would have restarted that run to ensure that if I could ;)

meanwhile addressing 

```
bundler/spec/dependabot/bundler/update_checker/force_updater_spec.rb:310:121: C: Layout/LineLength: Line is too long. [121/120]
    context "when peer dependencies in the Gemfile shouldn't update together, because one of them would be downgraded" do
```

via shorter word use

```
diff --git a/bundler/spec/dependabot/bundler/update_checker/force_updater_spec.rb b/bundler/spec/dependabot/bundler/update_checker/force_updater_spec.rb
index f9c9277e1..a1e8f6cae 100644
--- a/bundler/spec/dependabot/bundler/update_checker/force_updater_spec.rb
+++ b/bundler/spec/dependabot/bundler/update_checker/force_updater_spec.rb
@@ -307,7 +307,7 @@ RSpec.describe Dependabot::Bundler::UpdateChecker::ForceUpdater do
       end
     end
 
-    context "when peer dependencies in the Gemfile shouldn't update together, because one of them would be downgraded" do
+    context "when peer dependencies in the Gemfile shouldn't update together, since one of them would be downgraded" do
       let(:dependency_files) { bundler_project_dependency_files("no_downgrades") }
       let(:target_version) { "7.1.1" }
       let(:dependency_name) { "rails" }
```
advise otherwise if you see a better fit

**Comment by abdulapopoola** [comment]
Hi @yarikoptic ; sorry about the delay and really appreciate your enthusiasm and desire to make Dependabot much better. Do let me know what can be done to get your contributions merged.

**Comment by yarikoptic** [comment]
I have rebased now, adjusted config to skip more of urls, reran codespell to fix 2 new typos.  back in your field

**Review by abdulapopoola — APPROVED** [review]

**Review by brbayes-msft — APPROVED** [review]

File: updater/spec/dependabot/dependency_group_engine_spec.rb [hunk 2]
```diff
@@ -159,7 +159,7 @@
               - group-b
 
               This can happen if:
-              - the group's 'pattern' rules are mispelled
+              - the group's 'pattern' rules are misspelled
```
Comment (jurre) [inline updater/spec/dependabot/dependency_group_engine_spec.rb:162]: This was kind of poetic, imo 😆

**Comment by jurre** [comment]
The failures left in CI are now all unrelated to the changes in this PR

**Review by jurre — APPROVED** [review]

**Review by jurre — APPROVED** [review]

**Pull request merged**